### PR TITLE
MBS-12917: Fix title of credited entity’s link

### PR DESCRIPTION
**Pull request by yvanzo**
# Problem MBS-12917

Since the refactoring commit a252f37bd1de585cf3f87660cc5bee44eaea80e0 in beta only, the title shown on hover every link to any entity credited under a different name and rendered with React is broken, wrongly showing sort name for artist and nothing for entity of other type.

# Solution

- Improve test conditions instead of setting a default value to `nameVariation`
- Report any misuse of `nameVariation` depending on `content`’s type

# Testing

Tested reproducing the bug in a work relationship editor
Noticed that React links to credited entities have been fixed in many other parts of the website
Agreed on not adding further test for this specific issue as it would be reported in development or through Sentry at least.

### CODE REVIEW

**Comment by yvanzo** [comment]
@brainzbot, please retest this!

**Comment by mwiencek** [comment]
@brainzbot test this please

File: root/static/scripts/common/components/EntityLink.js
Comment (mwiencek) [inline root/static/scripts/common/components/EntityLink.js:204]: minor, but to remove newlines and whitespace from the message (for translated strings that's handled by `cleanMsgid`)
```suggestion
    const errorMessage = `Content of type ${typeof content} cannot be ` +
      'compared as a string to entity name for name variation.';
```

**Review by mwiencek — APPROVED** [review]
The test failure looks unrelated, so trying another run. Looks good to merge assuming those pass.

**Comment by reosarevok** [comment]
@brainzbot, retest this please

**Review by reosarevok — APPROVED** [review]
Tested locally with artists in several places (display, editor, editor popup). The code changes about display make sense to me. I haven't set up sentry capturing before, and I didn't test it here, so I defer to @mwiencek on that bit.

File: root/static/scripts/common/components/EntityLink.js
Comment (yvanzo) [reply root/static/scripts/common/components/EntityLink.js]: Good catch, now fixed: https://github.com/metabrainz/musicbrainz-server/compare/3e61be228c18742eae6d01be0fa504a8acc8d9f8..1a60e1fb39b41dc4088539feb5486151e10b427a

**Comment by yvanzo** [comment]
@brainzbot, please retest this, and make it pass…

**Comment by reosarevok** [comment]
@brainzbot, retest this please, it's tiresome already

**Comment by reosarevok** [comment]
@brainzbot, retest this please? Again?

**Comment by reosarevok** [comment]
@brainzbot, retest this please? :cry:

**Comment by yvanzo** [comment]
Rebased on latest head master which passed Selenium tests successfully.

**Comment by mwiencek** [comment]
It's perhaps a legitimate failure then. I can try running them locally to see.

File: root/static/scripts/common/components/EntityLink.js
Comment (mwiencek) [inline root/static/scripts/common/components/EntityLink.js:206]: This is triggering where content is `null`, which is causing the failures. Maybe check `nonEmpty(content)` instead of `content !== undefined`.

Comment (mwiencek) [reply root/static/scripts/common/components/EntityLink.js:206]: BTW, I was able to find the error like this:
 * Found which test the failing command `check target="xpath=//input[@id = 'change-matching-artists']" value=""` was in (it's in t/selenium/release-editor/The_Downward_Spiral.json5)
 * Ran that test with the browser open: `env SIR_DIR=/home/michael/code/sir ./t/selenium.js --headless=false --stay-open=true t/selenium/release-editor/The_Downward_Spiral.json5`
 * Opened the developer tools while the test was running and waited for an exception to occur

**Pull request merged**
